### PR TITLE
Added SetOptions support to RocksJava

### PR DESCRIPTION
**Pull request by adamretter**
Implemented `RocksDB#setOptions` and associated mutable/immutable options

### CODE REVIEW

File: java/rocksjni/options.cc [hunk 1]
```diff
@@ -1876,6 +1876,219 @@ void Java_org_rocksdb_Options_prepareForBulkLoad(
...
+  return reinterpret_cast<rocksdb::Options*>(
+      jhandle)->memtable_prefix_bloom_huge_page_tlb_size;
+}
+
+/*
+ * Class:     org_rocksdb_Options
+ * Method:    setMemtablePrefixBloomHugePageTlbSize
+ * Signature: (JJ)V
+ */
+void Java_org_rocksdb_Options_setMemtablePrefixBloomHugePageTlbSize(
```
Comment (siying) [inline java/rocksjni/options.cc]: I just renamed the option to memtable_huge_page_size.

**Comment by adamretter** [comment]
@siying I just adapted this for your change of `memtable_huge_page_size` and rebased it. Hopefully good to go now :-)

File: java/src/main/java/org/rocksdb/ColumnFamilyOptions.java [hunk 2]
```diff
@@ -776,6 +869,30 @@ private native void setMinPartialMergeOperands(
   private native void setOptimizeFiltersForHits(long handle,
       boolean optimizeFiltersForHits);
   private native boolean optimizeFiltersForHits(long handle);
+  private native void setMemtableHugePageSize(long handle,
+      long memtablePrefixBloomHugePageTlbSize);
```
Comment (lightmark) [inline java/src/main/java/org/rocksdb/ColumnFamilyOptions.java]: Could you also update the param name here to make them consistent? Thanks.

**Comment by adamretter** [comment]
@siying Done. Sorry about that, I did try and change them all, but somehow that one slipped through, it is fixed and rebased now.

**Pull request merged**